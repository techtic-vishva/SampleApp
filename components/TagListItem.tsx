import React from 'react'
import { StyleSheet, View } from 'react-native'
import { chattanoogaTapWater, dichotomousHippopotamus, fill } from '../constants/Colors'
import { formatDuration } from '../core/format'
import { SessionTagAgg } from '../core/services/user-stats'
import { Text } from './Themed'

export default function (props: { index: number; sessiogTagAgg: SessionTagAgg }) {
	return (
		<>
			<View style={styles.listViewContainer} key={props.index}>
				<View style={styles.listCounterView}>
					<Text style={styles.tagTxtStyle}>{props.index + 1}</Text>
				</View>
				<Text numberOfLines={1} style={[styles.tagTxtStyle, { flex: 1, marginStart: '5%' }]}>
					{props.sessiogTagAgg.name}
				</Text>
				<Text style={[styles.tagTxtStyle, { marginStart: '5%', color: chattanoogaTapWater, fontFamily: 'objektiv' }]}>
					{formatDuration(props.sessiogTagAgg.duration, 2)}
				</Text>
			</View>
		</>
	)
}

const styles = StyleSheet.create({
	listViewContainer: {
		height: 55,
		borderColor: fill,
		borderWidth: 3,
		alignItems: 'center',
		paddingStart: '5%',
		paddingEnd: '5%',
		marginBottom: 10,
		borderTopLeftRadius: 50,
		borderBottomLeftRadius: 50,
		borderTopRightRadius: 7,
		borderBottomRightRadius: 7,
		flexDirection: 'row'
	},
	listCounterView: {
		borderColor: dichotomousHippopotamus,
		borderWidth: 1,
		borderRadius: 20,
		height: 30,
		width: 30,
		alignItems: 'center',
		justifyContent: 'center'
	},
	tagTxtStyle: {
		color: 'white',
		fontFamily: 'objektiv-semi-bold',
		fontSize: 16
	}
})
