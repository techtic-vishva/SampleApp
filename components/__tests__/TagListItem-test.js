import * as React from 'react'
import renderer from 'react-test-renderer'

import TagListItem from '../TagListItem'

it('TagListItem renders correctly', () => {
	const tree = renderer
		.create(<TagListItem index={0} sessiogTagAgg={{ name: 'test', duration: { hours: '1' } }} />)
		.toJSON()
	expect(tree).toMatchSnapshot()
})
