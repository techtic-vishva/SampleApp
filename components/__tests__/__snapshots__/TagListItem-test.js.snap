// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagListItem renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomLeftRadius": 50,
      "borderBottomRightRadius": 7,
      "borderColor": "#23211e",
      "borderTopLeftRadius": 50,
      "borderTopRightRadius": 7,
      "borderWidth": 3,
      "flexDirection": "row",
      "height": 55,
      "marginBottom": 10,
      "paddingEnd": "5%",
      "paddingStart": "5%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#ed832d",
        "borderRadius": 20,
        "borderWidth": 1,
        "height": 30,
        "justifyContent": "center",
        "width": 30,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "objektiv",
          },
          Object {
            "color": "white",
            "fontFamily": "objektiv-semi-bold",
            "fontSize": 16,
          },
        ]
      }
    >
      1
    </Text>
  </View>
  <Text
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "#fff",
          "fontFamily": "objektiv",
        },
        Array [
          Object {
            "color": "white",
            "fontFamily": "objektiv-semi-bold",
            "fontSize": 16,
          },
          Object {
            "flex": 1,
            "marginStart": "5%",
          },
        ],
      ]
    }
  >
    test
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
          "fontFamily": "objektiv",
        },
        Array [
          Object {
            "color": "white",
            "fontFamily": "objektiv-semi-bold",
            "fontSize": 16,
          },
          Object {
            "color": "#999a81",
            "fontFamily": "objektiv",
            "marginStart": "5%",
          },
        ],
      ]
    }
  >
    1 hr
  </Text>
</View>
`;
